validphys2/src/validphys/covmats.py

```diff
@@ -535,9 +535,15 @@ def pdferr_plus_covmat(dataset, pdf, covmat_t0_considered):
     return pdf_cov + covmat_t0_considered
 
 
-def pdferr_plus_dataset_inputs_covmat(data, pdf, dataset_inputs_covmat_t0_considered):
+def pdferr_plus_dataset_inputs_covmat(data, pdf, dataset_inputs_covmat_t0_considered, fitthcovmat):
     """Like `pdferr_plus_covmat` except for an experiment"""
     # do checks get performed here?
+    if fitthcovmat is not None:
+        theory_covmat = fitthcovmat.load()
+        #change ordering according to exp_covmat (so according to runcard order)
+        tmp = theory_covmat.droplevel(0, axis=0).droplevel(0, axis=1)
+        bb = [str(i) for i in data]
+        return pdferr_plus_covmat(data, pdf, dataset_inputs_covmat_t0_considered+ tmp.reindex(index=bb, columns=bb, level=0).values)
```

Comment:
Could we have the complicated reindexing as a separate function somewhere?

Reply:
Yes, I am doing that